lib/logstash/inputs/file.rb

```diff
@@ -199,15 +199,15 @@ def run(queue)
         queue << event
       end
     end
-    finished
   end # def run
 
   public
-  def teardown
-    if @tail
-      @tail.sincedb_write
-      @tail.quit
-      @tail = nil
-    end
-  end # def teardown
+  def stop
+    super
+    @tail.quit if @tail # _sincedb_write is called implicitly
```

Comment:
I wonder how tail.quit handle this operation, does it waits to finish a subscribe call running? does this makes sense?